src/bin/download_sysext.rs

```diff
@@ -36,13 +38,51 @@ struct Package<'a> {
 
 impl<'a> Package<'a> {
     #[rustfmt::skip]
-    fn hash_on_disk(&mut self, path: &Path) -> Result<omaha::Hash<omaha::Sha256>, Box<dyn Error>> {
+    // Return Sha256 hash of data in the given path.
+    // If maxlen is None, a simple read to the end of the file.
+    // If maxlen is Some, read only until the given length.
+    fn hash_on_disk(&mut self, path: &Path, maxlen: Option<usize>) -> Result<omaha::Hash<omaha::Sha256>, Box<dyn Error>> {
         use sha2::{Sha256, Digest};
 
-        let mut file = File::open(path)?;
+        let file = File::open(path)?;
         let mut hasher = Sha256::new();
 
-        io::copy(&mut file, &mut hasher)?;
+        let filelen = file.metadata().unwrap().len() as usize;
+
+        let mut maxlen_to_read: usize = match maxlen {
+            Some(len) => {
+                if filelen < len {
+                    filelen
+                } else {
+                    len
+                }
+            }
+            None => filelen,
+        };
+
+        const CHUNKLEN: usize = 10485760; // 10M
+
+        let mut freader = BufReader::new(file);
+        let mut read_offset: usize = 0;
+        let mut chunklen: usize;
+
+        while maxlen_to_read > 0 {
```

Comment:
```suggestion
        freader.seek(SeekFrom::Start(0.try_into().unwrap()))?;
        while maxlen_to_read > 0 {
```
It's enough to seek to the start once.